### DISCUSSION TITLE
Problems starting Compose Quickstart of 3.0.0

### DISCUSSION THREAD

**Discussion by moley** [apiman/apiman#2487] [Help]
Hi anyone, I tried to start the quickstart of apiman like described here: https://www.apiman.io/download.html (Docker-Compose) but recieving the following error in browser navigating to http://apiman.local.gd:8080/apimanui: Bad Gateway.... I am living behind a proxy :-( Do you have any hints for me how to find out the problem?

**Comment by msavy** [comment]
Hi @moley, this isn't enough information to go on.

What operating system are you using? 

Do you see any errors in the docker compose logs?

**Comment by moley** [comment]
Hi @msavy , 

Sorry, yes I a m lacking some infos... 
I am trying on a MacOs 13.0.1
docker-compose -f docker-compose.setup.yml up returns OK, but docker-compose up contains following errors: 

`(MSC service thread 1-4) WFLYELY00024: Certificate [apimancert] in KeyStore is not valid: java.security.cert.CertificateExpiredException: NotAfter: Mon Apr 25 15:14:17 UTC 2016
apiman-manager_1    | 	at java.base/sun.security.x509.CertificateValidity.valid(CertificateValidity.java:277)
apiman-manager_1    | 	at java.base/sun.security.x509.X509CertImpl.checkValidity(X509CertImpl.java:675)
apiman-manager_1    | 	at java.base/sun.security.x509.X509CertImpl.checkValidity(X509CertImpl.java:648)
apiman-manager_1    | 	at org.wildfly.extension.elytron@15.0.1.Final//org.wildfly.extension.elytron.KeyStoreService.checkCertificatesValidity(KeyStoreService.java:230)
apiman-manager_1    | 	at org.wildfly.extension.elytron@15.0.1.Final//org.wildfly.extension.elytron.KeyStoreService.start(KeyStoreService.java:192)
apiman-manager_1    | 	at org.jboss.msc@1.4.12.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1739)
apiman-manager_1    | 	at org.jboss.msc@1.4.12.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1701)
apiman-manager_1    | 	at org.jboss.msc@1.4.12.Final//org.jboss.msc.service.ServiceControllerImpl$ControllerTask.run(ServiceControllerImpl.java:1559)
apiman-manager_1    | 	at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
apiman-manager_1    | 	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
apiman-manager_1    | 	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
apiman-manager_1    | 	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1363)
apiman-manager_1    | 	at java.base/java.lang.Thread.run(Thread.java:834)`

followed by a lot of IndexOutOfBoundsExceptions like

`19:53:50,653 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-8) WFLYSRV0003: Could not index class com/google/common/util/concurrent/AbstractListeningExecutorService.class at /content/apiman-developer-portal-wildfly.war/WEB-INF/lib/guava-31.1-jre.jar: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
apiman-manager_1    | 	at org.jboss.jandex@2.2.3.Final//org.jboss.jandex.Indexer.updateTypeTarget(Indexer.java:903)
apiman-manager_1    | 	at org.jboss.jandex@2.2.3.Final//org.jboss.jandex.Indexer.updateTypeTargets(Indexer.java:630)
apiman-manager_1    | 	at org.jboss.jandex@2.2.3.Final//org.jboss.jandex.Indexer.index(Indexer.java:1698)
apiman-manager_1    | 	at org.jboss.as.server@15.0.1.Final//org.jboss.as.server.deployment.annotation.ResourceRootIndexer.indexResourceRoot(ResourceRootIndexer.java:99)
apiman-manager_1    | 	at org.jboss.as.server@15.0.1.Final//org.jboss.as.server.deployment.annotation.AnnotationIndexProcessor.deploy(AnnotationIndexProcessor.java:51)
apiman-manager_1    | 	at org.jboss.as.server@15.0.1.Final//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:182)
apiman-manager_1    | 	at org.jboss.msc@1.4.12.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1739)
apiman-manager_1    | 	at org.jboss.msc@1.4.12.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1701)
apiman-manager_1    | 	at org.jboss.msc@1.4.12.Final//org.jboss.msc.service.ServiceControllerImpl$ControllerTask.run(ServiceControllerImpl.java:1559)
apiman-manager_1    | 	at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
apiman-manager_1    | 	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
apiman-manager_1    | 	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
apiman-manager_1    | 	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1363)
apiman-manager_1    | 	at java.base/java.lang.Thread.run(Thread.java:834)
apiman-manager_1    |`

A docker logs of the traefix container also resulted in 
`time="2023-03-21T19:53:58Z" level=error msg="accept tcp [::]:8443: use of closed network connection" entryPointName=https
time="2023-03-21T19:53:58Z" level=error msg="Error while starting server: accept tcp [::]:8443: use of closed network connection" entryPointName=https
time="2023-03-21T19:53:58Z" level=error msg="accept tcp [::]:8080: use of closed network connection" entryPointName=http
time="2023-03-21T19:53:58Z" level=error msg="close tcp [::]:8080: use of closed network connection" entryPointName=http`


Don't know if one of these errors close the gap, else please request more infos

Thanks in advance

**Reply by msavy** [reply]
1. Self-signed keystore being expired is normal. For production, you would create your own.
2. `IndexOutOfBoundsException` warnings can be ignored, these are from Jandex; it can't index some Guava classes. This will go away in a future version of Apiman when we move to a newer version of WildFly.
3. Not immediately clear what is causing that. Seems like something is cutting off the connections.

Are you unable to resolve `local.gd` via DNS because of your workplace DNS or proxy or something?

The Docker Compose quickstart is just meant for people to quickly try out Apiman on their development machines, so we use `local.gd` to it possible to have subdomains without needing a "real" set of DNS domain names.

**Reply by moley** [reply]
Yes,it seems so, because I am getting a fancy DNS screen in the browser. I think that local.gd is detectd as external network. I think I will reach out to network collegues of my company first to try to handle this locally

**Reply by msavy** [reply]
At least on macOS, this works fine for me:

- `docker compose down`
- Edit `.env`
- Change `APIMAN_HOSTNAME` to `localhost`
- `docker compose up`

Then your URL will be something like `http://apiman.localhost:8080/apimanui/`.

**Reply by msavy** [reply]
I hope this worked OK for you. Let me know how things went.
